### DISCUSSION TITLE
Use `fastlane match appstore`   command without apple account password

### DISCUSSION THREAD

**Discussion by AcarFurkan** [fastlane/fastlane#22245] [Q&A]
Can I log in to fastlane match without apple password with API key?

I run the `fastlane match appstore` command and it asks for the repo and I give the link. Then it asks for the information of the apple account. Is there an option that I can continue by giving an api key instead of username and password? For example, we use google auth and almost no one has a password.

**Comment by bella964** [comment]
Hello,
Yes, you can authenticate fastlane match using an API key instead of your Apple ID username and password. Apple introduced App Store Connect API keys, which you can use for automated workflows like this.    [Read Works
](https://www-readworks.com)
To authenticate using the API key, follow these steps:

Generate the API Key in App Store Connect:

Go to App Store Connect.
Under Users and Access, click Keys and create an API key. This will generate a key ID, issuer ID, and the private key (a .p8 file).
Configure Fastlane to use the API Key:

Download the API key file (.p8) and save it somewhere secure.

Set the following environment variables in your terminal to authenticate with the API key:

bash
export FASTLANE_APPLE_ID="your_apple_id"  # Optional if you don't specify in the fastlane configuration
export FASTLANE_APPLE_API_KEY_PATH="/path/to/AuthKey_XXXXXX.p8"
export FASTLANE_APPLE_API_KEY_ISSUER_ID="your_issuer_id"
export FASTLANE_APPLE_API_KEY_ID="your_key_id"
Run fastlane match:

Now you can run fastlane match without being prompted for a password:

bash
fastlane match appstore --type appstore
With this setup, fastlane will authenticate using the API key instead of prompting for an Apple ID password.

Best Regards
bella964
